### PR TITLE
Document how to prefer local networks

### LINKED ISSUE #3256: Prefer local networks when routing traffic
Background: https://airtable.com/tblD3oZPLJgGhCmch/viw8IfZMg9Up8SP33/recmrMuwQEfQ6JiIq
Eng partner: @bdarnell 
PM: @awoods187

### PR DESCRIPTION
**Pull request by jseldess**
- Add `--locality-advertise-addr` flag to `cockroach start` docs
- Feature the flag in the Networking section of the Production Checklist

Fixes #3256.

### CODE REVIEW

File: v2.1/start-a-node.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ Flag | Description
...
+`--locality-advertise-addr` | <span class="version-tag">New in v2.1:</span> The IP address/hostname and port to tell other nodes in specific localities to use. This flag is useful when running a cluster across multiple networks, where nodes in a given network have access to a private or local interface while nodes outside the network do not. In this case, you can can use `--locality-advertise-addr` to tell nodes within the same network to prefer the private or local address to improve performance and use `--advertise-addr` to tell nodes outside the network to use another address that is reachable from them.<br><br>This feature relies on nodes being started with the [`--locality`](#locality) flag. The flag should be specified as a commma-separated list of `locality@address` values, for example:<br><br>`--locality-advertise-addr=region=us-west@127.0.0.1:26257,datacenter=us-west-1b@127.0.0.1:26258`<br><br>See the [example](#start-a-multi-node-cluster-across-private-networks) below for more details.
```
Comment (bdarnell) [inline v2.1/start-a-node.md]: 127.0.0.1 is not a realistic example here. I'd use different addresses in 10.0.0.0/8:

`--locality-advertise-addr=region=us-west@10.1.2.3:26257,region=us-east@10.5.6.7:26257`

(While comma-separated lists are theoretically possible here, I'm not sure if they'd ever be used in practice. In the overwhelming majority of cases, a node will have one entry in `locality-advertise-addr` for one of its own locality fields)

Comment (jseldess) [reply v2.1/start-a-node.md]: Thanks, @bdarnell. Fixed. Though I used `10.0.0.0:26257` in the example, not `10.0.0.0/8:26257`. That ok? Wasn't clear on that.

Comment (bdarnell) [reply v2.1/start-a-node.md]: Right, I didn't mean for you to use `/8`. `10.0.0.0/8` describes the private network in use; addresses in that network don't have the `/8`.

File: v2.1/recommended-production-settings.md [hunk 2]
```diff
@@ -144,7 +143,7 @@ When running a cluster across multiple networks, the setup depends on whether no
...
+No | This is typical when networks are on different clouds. In this case, set up a [VPN](https://en.wikipedia.org/wiki/Virtual_private_network), [VPC](https://en.wikipedia.org/wiki/Virtual_private_cloud), [NAT](https://en.wikipedia.org/wiki/Network_address_translation), or another such solution to provide unified routing across the networks. Then start each node with `--advertise-addr` set to the address that is reachable from other networks and do not specify `--listen-addr`. This will tell other nodes to use the specific IP address advertised, but load balancers/clients will be able to use any address that routes to the node.<br><br><span class="version-tag">New in v2.1:</span> Also, if a node is reachable from other nodes in its network on a private or local address, set `--locality-advertise-addr` to that address. This will tell nodes within the same network to prefer the private or local address to improve performance. Note that this feature requires that each node is started with the [`--locality`](start-a-node.html#locality) flag. For more details, see this [example](start-a-node.html#start-a-multi-node-cluster-across-private-networks).
```
Comment (sploiselle) [inline v2.1/recommended-production-settings.md]: I might be misunderstanding this, but it seems like this feature should be the first recommendation in the `No` section? If there's something I'm missing here, please disregard.

Comment (sploiselle) [reply v2.1/recommended-production-settings.md]: >  set `--locality-advertise-addr` to that address.

Is there a good place we could create a link here?

**Comment by sploiselle** [comment]
@jseldess Two small questions, but LGTM mod those.

File: v2.1/recommended-production-settings.md (hunk 2 above)
Comment (jseldess) [reply v2.1/recommended-production-settings.md]: TFTR, @sploiselle. 

Added a link to the `Networking` flags. 

In terms of placing the feature, I think it's still important to start by focusing on how to get the networks to talk at all (VPN/NAT, etc.). If this isn't possible, the `--locality-advertise-addr` feature won't matter much.

**Pull request merged**